solvebio/cli/data.py

```diff
@@ -159,10 +162,63 @@ def _upload_folder(
                     )
                 )
             else:
-                remote_parent = Object.get_by_full_path(
-                    remote_folder_full_path, assert_type="folder"
-                )
-                Object.upload_file(local_file_path, remote_parent.path, vault.full_path)
+                all_files.append((local_file_path, remote_folder_full_path, vault.full_path))
+
+    # Identify which remote folders already exist
+    upload_root_path, _ = Object.validate_full_path(
```

Comment:
Would it be possible to only call this code block if num processes > 1? just for backwards compatibility?

Reply:
Absolutely. Can make this change.